packages/x-charts/src/ChartContainer/useChartContainerProps.ts

```diff
@@ -5,15 +5,19 @@ import { ChartDataProviderProps } from '../ChartDataProvider';
 import type { ChartContainerProps } from './ChartContainer';
 import { ChartSeriesType } from '../models/seriesType/config';
 import { ALL_PLUGINS, AllPluginSignatures, AllPluginsType } from '../internals/plugins/allPlugins';
+import { ChartAnyPluginSignature } from '../internals/plugins/models';
 
 export type UseChartContainerPropsReturnValue<TSeries extends ChartSeriesType> = {
   chartDataProviderProps: ChartDataProviderProps<TSeries, AllPluginSignatures<TSeries>>;
   chartsSurfaceProps: ChartsSurfaceProps & { ref: React.Ref<SVGSVGElement> };
   children: React.ReactNode;
 };
 
-export const useChartContainerProps = <TSeries extends ChartSeriesType = ChartSeriesType>(
-  props: ChartContainerProps<TSeries>,
+export const useChartContainerProps = <
+  TSeries extends ChartSeriesType = ChartSeriesType,
+  TSignatures extends readonly ChartAnyPluginSignature[] = AllPluginSignatures<TSeries>,
```

Comment:
Do we need to be strict in these plugin types? Can't we simply have a type that accepts everything for these generic places and use a subset of the full type where it is needed? 🤔

Reply:
I don't get your question. Which typing could be more flexible?

Reply:
It seems that we need to define what plugins the `useChartContainerProps` function should use/return but it doesn't really care much about them, and we have to force some types with `as`. 

This probably means our types are either too flexible or not flexible enough, my comment was more of a thought provoking one, to see if you have other solution for this.